### PR TITLE
Changed filename to use pk as descriptor  (bug 1205349) 

### PR DESCRIPTION
**Pull request by Nolski**
There are a few issues here. The file that is displayed in the UI doesn't seem to have any relation to the filename that we display here. I'm not sure if this is okay or not. pk doesn't seem to always have a value in the tests which I suppose is okay.

### CODE REVIEW

**Comment by andymckay** [comment]
There a bug number for this?

File: apps/files/models.py [hunk 1]
```diff
@@ -597,7 +597,7 @@ def save(self, *args, **kw):
         super(FileUpload, self).save()
 
     def add_file(self, chunks, filename, size):
-        filename = smart_str(filename)
+        filename = smart_str('{0}_{1}'.format(self.pk, filename))
```
Comment (andymckay) [inline apps/files/models.py]: do we need to do anything on existing files?

Comment (Nolski) [inline apps/files/models.py]: As far as I can tell we shouldn't seeing as this doesn't seem to affect how we currently search for files. The only use of using the file name as a lookup was in the unit tests (which is not fixed). @magopian may have a better idea however.

Comment (magopian) [inline apps/files/models.py]: Yeah, I don't believe we're using the filename on disk in a hardcoded way anywhere. So changing it shouldn't cause any issues, and it'll only be for the newly uploaded files.

File: apps/devhub/tests/test_views.py [hunk 2]
```diff
@@ -1974,8 +1974,8 @@ def test_login_required(self):
     def test_create_fileupload(self):
         self.post()
 
-        upload = FileUpload.objects.get(name='animated.png')
-        eq_(upload.name, 'animated.png')
+        upload = FileUpload.objects.filter().order_by('-created')[0]
```
Comment (magopian) [inline apps/devhub/tests/test_views.py]: You should be able to use `.first()` or `.last()`

File: apps/devhub/tests/test_views.py [hunk 3]
```diff
@@ -1974,8 +1974,8 @@ def test_login_required(self):
...
+        assert 'animated.png' in upload.name
```
Comment (magopian) [inline apps/devhub/tests/test_views.py]: You can test the filename for real: `assert '{0}_animated.png'.format(upload.pk)`

File: apps/devhub/tests/test_views.py [hunk 4]
```diff
@@ -2075,12 +2075,12 @@ def test_detail_json(self):
...
     def test_detail_view(self):
         self.post()
-        upload = FileUpload.objects.get(name='animated.png')
+        upload = FileUpload.objects.filter().order_by('-created')[0]
         r = self.client.get(reverse('devhub.upload_detail',
                                     args=[upload.uuid]))
         eq_(r.status_code, 200)
         doc = pq(r.content)
-        eq_(doc('header h2').text(), 'Validation Results for animated.png')
+        eq_(doc('header h2').text(), 'Validation Results for _animated.png')
```
Comment (magopian) [inline apps/devhub/tests/test_views.py]: usually I change the `eq_` to `assert` whenever I touch a test.

File: apps/files/models.py (hunk 1 above)
Comment (magopian) [inline apps/files/models.py]: You need to make sure the `FileUpload` is already saved, or `self.pk` will be empty. Two solutions: either you add a `fu.save()` after https://github.com/mozilla/olympia/pull/769/files#diff-7962dc902d031e550d31c50d17f2ca20R618, or even better, add the following just above:
```
if not self.pk:
    self.save()
```

**Comment by magopian** [comment]
This is looking good! There's two test failures that are caused by testing against the hardcoded name without the pk I believe.

**Comment by andymckay** [comment]
It looks like those tests have been resolved @magopian ?

**Comment by magopian** [comment]
indeed, r+

**Comment by andymckay** [comment]
@Nolski I think you can merge this with the r+ from @magopian.

**Pull request merged**